src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerValidatorTest.kt

```diff
@@ -0,0 +1,50 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonersearch
+
+import jakarta.validation.ValidationException
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.stub
+import org.mockito.kotlin.whenever
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.BIRMINGHAM
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.MOORLAND
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.helper.isEqualTo
+
+class PrisonerValidatorTest {
+
+  private val prisoner: Prisoner = mock()
+  private val client: PrisonerSearchClient = mock()
+  private val validator = PrisonerValidator(client)
+
+  @Test
+  fun `should pass validation when prisoner found`() {
+    prisoner.stub { on { prisonId } doReturn BIRMINGHAM }
+
+    whenever(client.getPrisoner("123456")) doReturn prisoner
+
+    val returnedPrisoner = validator.validatePrisonerAtPrison("123456", BIRMINGHAM)
+
+    returnedPrisoner isEqualTo prisoner
+  }
+
+  @Test
+  fun `should fail validation when prisoner found but not at prison`() {
+    prisoner.stub { on { prisonId } doReturn MOORLAND }
+
+    whenever(client.getPrisoner("123456")) doReturn prisoner
+
+    val error = assertThrows<ValidationException> { validator.validatePrisonerAtPrison("123456", BIRMINGHAM) }
+
+    error.message isEqualTo "Prisoner 123456 not found at prison BMI"
```

Comment:
Might be OUT, but only temporarily?

Reply:
So I have not really gone to far with validating a prisoners whereabouts.  It gets very complicated and we probably want to decide how far we need to go.

All we are saying at the moment is the prisoner at the prison.  They can still be active in or active out here.  We are not asking if they are in the prison.

Reply:
OK sounds good. I was thinking we just need an indication they are at the prison, and that the apppointment location is valid for the prison, and not much more beyond that.